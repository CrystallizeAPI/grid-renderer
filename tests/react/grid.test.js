import React from 'react';
import { shallow } from 'enzyme';
import Grid from '../../src/react/grid';
import { CustomCellComponent, model, toCells } from './utils';

const { rows } = model;
const cells = toCells(rows);

describe('Grid', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Grid cells={cells} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a Cell component for each cell', () => {
    const wrapper = shallow(<Grid cells={cells} />);
    const cellComponents = wrapper.find('Cell');
    expect(cellComponents).toHaveLength(3);
    expect(cellComponents.first().prop('cell')).toEqual(rows[0].columns[0]);
  });

  describe('when the `cellComponent` prop is given', () => {
    it('renders the custom component for each cell', () => {
      const wrapper = shallow(
        <Grid cellComponent={CustomCellComponent} cells={cells} />
      );
      const cellComponents = wrapper.find(CustomCellComponent);
      expect(cellComponents).toHaveLength(3);
      expect(cellComponents.first().prop('cell')).toEqual(rows[0].columns[0]);
    });
  });

  describe('when the `renderCellContent` prop is given', () => {
    it('calls the prop as a function for each cell', () => {
      const renderCellContent = jest.fn(cell => (
        <CustomCellComponent cell={cell} />
      ));
      const wrapper = shallow(
        <Grid renderCellContent={renderCellContent} cells={cells} />
      );
      expect(renderCellContent.mock.calls).toHaveLength(3);
      expect(wrapper.find(CustomCellComponent)).toHaveLength(3);
    });
  });
});
