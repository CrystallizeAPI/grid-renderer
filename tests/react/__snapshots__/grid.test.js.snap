// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid renders correctly 1`] = `
<styled.div
  totalColSpan={4}
>
  <Cell
    cell={
      Object {
        "item": Object {
          "name": "Item 1",
        },
        "layout": Object {
          "colspan": 2,
          "rowspan": 1,
        },
      }
    }
    key="cell-0"
  />
  <Cell
    cell={
      Object {
        "item": Object {
          "name": "Item 2",
        },
        "layout": Object {
          "colspan": 2,
          "rowspan": 1,
        },
      }
    }
    key="cell-1"
  />
  <Cell
    cell={
      Object {
        "item": Object {
          "name": "Item 3",
        },
        "layout": Object {
          "colspan": 4,
          "rowspan": 1,
        },
      }
    }
    key="cell-2"
  />
</styled.div>
`;
